coreneuron/io/reports/report_handler.cpp

```diff
@@ -13,18 +13,40 @@
 
 namespace coreneuron {
 
-void ReportHandler::create_report(double dt, double tstop, double delay) {
+std::vector<uint64_t> intersection_gids(const NrnThread& nt, std::vector<int>& target_gids) {
+    std::vector<int> thread_gids;
+    for (int i = 0; i < nt.ncell; i++) {
+        thread_gids.push_back(nt.presyns[i].gid_);
+    }
+    std::vector<uint64_t> intersection;
```

Comment:
as this is for saving memory - as vectors are int, `intersection` can be also int instead of uint64_t? (but I assume you used the later for future proofing)

Reply:
`uint32_t` you mean? :)

Reply:
make that template function and return with the same type as the arguments 😎